Library/Formula/open-mpi.rb

```diff
@@ -29,6 +30,7 @@ class OpenMpi < Formula
 
   depends_on :fortran => :recommended
   depends_on "libevent"
+  depends_on :java if build.with? "mpi-java"
```

Comment:
You can change this to `depends_on :java => :optional` and it'll create an option automatically (something like "Build with Java support"), then you don't need line 25 and line 49 can be `if build.with? "java"`.